fish_prompt.fish

```diff
@@ -115,34 +127,44 @@ function pre_prompt --on-event fish_prompt
       end
 
       if test $git_arrow_right != 0
-        set git_arrows " $git_arrows$pure_symbol_git_down_arrow"
+        set git_arrows "$git_arrows$pure_symbol_git_down_arrow"
       end
     end
 
     # Format Git prompt output
-    set pre_prompt $pre_prompt "$pure_color_gray$git_branch_name$git_dirty$pure_color_normal$pure_color_cyan$git_arrows$pure_color_normal "
-  end
-
-  if test $pure_user_host_location -ne 1
-    set pre_prompt $pre_prompt $user_and_host
+    set git_branch_f "$pure_color_gray"
+    set git_arrows_f "$pure_color_cyan"
   end
 
   # Prompt command execution duration
   if test -n "$CMD_DURATION"
-    set command_duration (__format_time $CMD_DURATION $pure_command_max_exec_time)
+    set cmd_duration (__format_time $CMD_DURATION $pure_command_max_exec_time)
   end
 
-  set pre_prompt $pre_prompt "$pure_color_yellow$command_duration$pure_color_normal"
+  set cmd_duration_f "$pure_color_yellow"
+
+  # Format current folder on prompt output
+  set cur_folder "" (__parse_current_folder)
+  set cur_folder_f "$pure_color_blue"
+
+  # Get the length of the first line
+  set -l pre_len (string length "$cur_folder[-1]$git_branch$git_dirty$git_arrows$user$at$host$cmd_duration")
+
+  if test $pre_len -gt $COLUMNS
+    # First line is too long, use the shortened format for current folder
+    set cur_folder "" (prompt_pwd)
+  end
+
+  if test $pure_user_host_location -eq 0
+    set pre_prompt "$pref$cur_folder_f$cur_folder[-1]$git_branch_f$git_branch$git_dirty_f$git_dirty$git_arrows_f$git_arrows$user_f$user$at_f$at$host_f$host$cmd_duration_f$cmd_duration"
```

Comment:
We can improve this assignment readibility/maintanability by using array syntax:

First declare with multi-line:

```
set pre_prompt \
          $pref \
          $cur_folder_f \
          $cur_folder[-1] \
          $git_branch_f \
          $git_branch
          ...
```
Then join

    set pre_prompt (string join '' $pre_prompt)

It will be easier to understand how `pre_prompt` is built and remove some pain for the future :heart: